### PR TITLE
Add keys to browse back and forward in help-mode

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Implemented https://github.com/syl20bnr/spacemacs/issues/2698

and + follow symbol with <kbd>s</kbd>

### CODE REVIEW

**Comment by StreakyCobra** [comment]
Maybe using (also) `n` and `N`, and/or `u` and `d` can be useful here? Not sure though.

**Comment by AdrieanKhisbe** [comment]
@StreakyCobra   <kbd>u</kbd> and <kbd>d</kbd> are undefined.what do you want me to bind them to? :)

<kbd>N/n</kbd> are already bind to `evil-search-next`

**Comment by StreakyCobra** [comment]
I was thinking that, as people are used to `C-u` and `C-d` to scroll, `u` (up) and `d` (down) would be kind of mnemonics. But I just tried and `C-u` and `C-d` are working so, it may not be necessary at all to add bindings?

On the other hand if we really want single-key scrolling, `f`, `b`, `n` and `N` are already used by evil, so it may be wise to use `u` and `d` instead of `f` and `b`. When in help buffers, I sometime want to select region (to copy/paste for instance) and using `b` to go to previous word, or `f` to find a symbol, are quite useful. So what do you think of using `u` and `d` instead, as:

- `d` is not useful because help buffers are read-only so it's not possible to delete line; and
- `u` for the same reason as there is nothing to undo because of read-only?

**Comment by TheBB** [comment]
These functions don't scroll, they go back and forward between pages, like in a browser.

**Comment by TheBB** [comment]
In info the `]` and `[` keys go forward and back. I like those.

**Comment by StreakyCobra** [comment]
Yeah, as I said in my previous (wrong) comment, I use `f` and `b` sometimes and I'm probably not the only one, so it may be wise to find better bindings.

**Comment by AdrieanKhisbe** [comment]
I mean *keys*

**Comment by StreakyCobra** [comment]
I'm not sure `f` and `b` should be used as the override some useful evil bindings, @TheBB What do you think?

`u` and `d` are fine for me. I can live with `C-u` and `C-d` only though.

**Comment by AdrieanKhisbe** [comment]
@StreakyCobra I didn't touch to `f` and `b` for these reasons

File: layers/+distribution/spacemacs-base/keybindings.el [hunk 1]
```diff
@@ -50,6 +50,14 @@
 ;; let TAB jump between links in help buffers
 (evil-define-key 'motion help-mode-map (kbd "<tab>") 'forward-button)
 (evil-define-key 'motion help-mode-map (kbd "S-<tab>") 'backward-button)
+;; let n and p to navigate between pages
+(evil-define-key 'motion help-mode-map (kbd "f") 'help-go-forward)
+(evil-define-key 'motion help-mode-map (kbd "b") 'help-go-back)
```
Comment (StreakyCobra) [inline layers/+distribution/spacemacs-base/keybindings.el]: You are adding `f` and `b` here no? Or did i missed something?

Comment (AdrieanKhisbe) [inline layers/+distribution/spacemacs-base/keybindings.el]: Well, forgot I made a mistake as it seems: cf the comment :D

File: layers/+distribution/spacemacs-base/keybindings.el [hunk 2]
```diff
@@ -50,6 +50,14 @@
 ;; let TAB jump between links in help buffers
 (evil-define-key 'motion help-mode-map (kbd "<tab>") 'forward-button)
 (evil-define-key 'motion help-mode-map (kbd "S-<tab>") 'backward-button)
+;; let n and p to navigate between pages
+(evil-define-key 'motion help-mode-map (kbd "n") 'help-go-forward)
+(evil-define-key 'motion help-mode-map (kbd "p") 'help-go-back)
```
Comment (StreakyCobra) [inline layers/+distribution/spacemacs-base/keybindings.el]: But now `n` is also used by evil for searching :stuck_out_tongue_winking_eye: 

Simply keep `[` and `]` for sure, and `s`, `u` and `d` if you think they are useful :relaxed:

Comment (AdrieanKhisbe) [inline layers/+distribution/spacemacs-base/keybindings.el]: indeed. ^^

**Comment by AdrieanKhisbe** [comment]
modified and accordingly, and squashed commit

**Comment by StreakyCobra** [comment]
:+1:

**Comment by AdrieanKhisbe** [comment]
:)

**Comment by syl20bnr** [comment]
I'm not sure we need `u` and `d`, there is already `C-u` and `C-d` which are consistent accross Emacs. I find it not really useful to create a specificity in help mode.
Follow-symbol should be on `mhh` (major-mode binding) to be consistent with the conventions, `s` is useful for people using evil-sniper.
Also for forward and back I propose the additional `gf` and `gb`.

**Comment by AdrieanKhisbe** [comment]
@syl20bnr Applied your recommandations :)
(I still have to lean vim conventions :D)

**Comment by syl20bnr** [comment]
Thank you :+1: 
I replaced `mhh` by `gh` and added the documentation where I could, see commit https://github.com/syl20bnr/spacemacs/commit/5895e0e
Cherry-picked into develop branch, you can safely delete your branch.

**Comment by AdrieanKhisbe** [comment]
@syl20bnr  great :)

**Pull request closed without merging**